mithril-common/src/entities/cardano_database.rs

```diff
@@ -106,6 +122,109 @@ pub struct ArtifactsLocations {
 #[typetag::serde]
 impl Artifact for CardanoDatabaseSnapshot {
     fn get_id(&self) -> String {
-        self.merkle_root.clone()
+        self.hash.clone()
+    }
+}
+
+#[cfg(test)]
+mod tests {
+    use super::*;
+
+    fn dummy() -> CardanoDatabaseSnapshot {
+        CardanoDatabaseSnapshot::new(
+            "mk-root-1111111111".to_string(),
+            CardanoDbBeacon::new(2222, 55555),
+            0,
+            ArtifactsLocations {
+                digests: vec![],
+                immutables: vec![],
+                ancillary: vec![],
+            },
+            CompressionAlgorithm::Gzip,
+            &Version::new(1, 0, 0),
+        )
+    }
+
+    #[test]
+    fn test_cardano_database_snapshot_compute_hash() {
+        let cardano_database_snapshot = CardanoDatabaseSnapshot {
+            merkle_root: "mk-root-123".to_string(),
+            beacon: CardanoDbBeacon::new(123, 98),
+            ..dummy()
+        };
+
+        assert_eq!(
+            "b1cc5e0deaa7856e8e811e349d6e639fa667aa70288602955f438c5893ce29c8",
+            cardano_database_snapshot.compute_hash()
+        );
+    }
+
+    #[test]
+    fn compute_hash_returns_same_hash_with_same_cardano_database_snapshot() {
+        assert_eq!(
+            CardanoDatabaseSnapshot {
+                merkle_root: "mk-root-123".to_string(),
+                beacon: CardanoDbBeacon::new(123, 98),
+                ..dummy()
+            }
+            .compute_hash(),
+            CardanoDatabaseSnapshot {
+                merkle_root: "mk-root-123".to_string(),
+                beacon: CardanoDbBeacon::new(123, 98),
+                ..dummy()
+            }
+            .compute_hash()
+        );
+    }
+
+    #[test]
+    fn compute_hash_returns_different_hash_with_different_merkle_root() {
+        assert_ne!(
+            CardanoDatabaseSnapshot {
+                merkle_root: "mk-root-123".to_string(),
+                beacon: CardanoDbBeacon::new(123, 98),
+                ..dummy()
+            }
+            .compute_hash(),
+            CardanoDatabaseSnapshot {
+                merkle_root: "mk-root-456".to_string(),
+                beacon: CardanoDbBeacon::new(123, 98),
+                ..dummy()
+            }
+            .compute_hash()
+        );
+    }
+
+    #[test]
+    fn compute_hash_returns_different_hash_with_different_beacon() {
```

Comment:
Return different hash with different beacon epoch but return same hash with different immutable file.
We may create 2 tests to give them a more explicit name

Reply:
Done :+1: